content/geometry/ConvexHull.h

```diff
@@ -24,22 +24,17 @@ Points on the edge of the hull between two other points are not considered part
 #include "Point.h"
 
 typedef Point<ll> P;
-pair<vi, vi> ulHull(const vector<P>& S) {
-	vi Q(sz(S)), U, L;
-	iota(all(Q), 0);                                                                                                                                                                         
-	sort(all(Q), [&S](int a, int b){ return S[a] < S[b]; }); 
-	trav(it, Q) {
-#define ADDP(C, cmp) while (sz(C) > 1 && S[C[sz(C)-2]].cross(\
-	S[it], S[C.back()]) cmp 0) C.pop_back(); C.push_back(it);
-		ADDP(U, <=); ADDP(L, >=);
-	}   
-	return {U, L}; 
-}
-
-vi convexHull(const vector<P>& S) {
-	vi u, l; tie(u, l) = ulHull(S);
-	if (sz(S) <= 1) return u;
-	if (S[u[0]] == S[u[1]]) return {0};
-	l.insert(l.end(), u.rbegin()+1, u.rend()-1);
-	return l;
-}
+vector<P> convexHull(vector<P> pts) {
+	if (sz(pts) <= 1) return pts;
+	sort(all(pts));
+	vector<P> h(sz(pts)+1);
+	int t = 0;
+	for (int it = 0, s = t; it < 2; it++, s = t) {
```

Comment:
`s = --t` at the end?

Reply:
:+1: 

I can also fit in the `reverse` in the loop.